Helper/PaymentResponseHandler.php

```diff
@@ -217,13 +230,28 @@ public function handlePaymentResponse($paymentsResponse, $payment, $order = null
                     $order->setActionFlag(\Magento\Sales\Model\Order::ACTION_FLAG_CANCEL, true);
 
                     if ($order->canCancel()) {
-                        $order->cancel();
-                        $order->save();
+                        if ($this->orderManagement->cancel($order->getEntityId())) { //new canceling process
+                            try {
+                                $orderStatusHistory = $this->orderStatusHistoryFactory->create()
+                                    ->setParentId($order->getEntityId())
+                                    ->setEntityName('order')
+                                    ->setStatus(Order::STATE_CANCELED)
+                                    ->setComment(__('Order has been cancelled by "%1" payment response.', $payment->getMethod()));
+                                $this->orderManagement->addComment($order->getEntityId(), $orderStatusHistory);
+                        } catch (\Exception $e) {
+                                $this->adyenLogger->addAdyenDebug(
+                                        __('Order cancel history comment error: %1', $e->getMessage())
+                                    );
+                                }
```

Comment:
Small formatting fix: 
```suggestion
                            } catch (\Exception $e) {
                                $this->adyenLogger->addAdyenDebug(
                                    __('Order cancel history comment error: %1', $e->getMessage())
                                );
                            }
```